cross/mingw-w64/Portfile

```diff
@@ -150,16 +150,11 @@ if {${subport} ne ${name}} {
     destroot.violate_mtree  yes
     livecheck.type          none
 } else {
+    PortGroup               stub 1.0
+
+    stub.noarch             yes
```

Comment:
Why is this not simply the default, and ports that actually need to depend on architecture would explicitly set this to `no` inestad? Isn't the default for most stub ports to just install a README file?

Reply:
> Why is this not simply the default, and ports that actually need to depend on architecture would explicitly set this to `no` inestad? Isn't the default for most stub ports to just install a README file?
> 
> Similar question for `install_libs`. I didn't explicitly set it to `false` because I never thought of it.

Per discussion with @jmroot, we felt it would be safer to use defaults that don't materially change how dependencies and such work. Does that make sense?

Reply:
Not too much if the portgroup is actually meant to handle ports that are fundamentally different from any other ports anyway.

We have just a handful of ports for now, so we can easily make sure that the variables are set correctly on all of them.

I'm not sure what's the exact intent of the portgroup, but my first impression is that it should handle any ports that install just a README file and maybe add a dependency or two. Nothing else. For those cases I would also expect reasonable defaults for this scenario, and an easy way out (setting a single variable to `true` or `false`) in case the port needs something else.

What I do find problematic are portgroups that provide 90 % of the needed functionality, but then set one fatal variable that cannot easily be reverted when you don't want that part. (One example that comes to mind is using the `python` PG for compiling some C++ software. They provide tons of nice helpers, but setting the configure/build to the default port behaviour is almost impossible. But that's different from this case.)

Reply:
> Not too much if the portgroup is actually meant to handle ports that are fundamentally different from any other ports anyway.
> 
> We have just a handful of ports for now, so we can easily make sure that the variables are set correctly on all of them.

For sure! And I'd be fine with enabling `noarch` by default, given that it's enabled for the vast majority of our stubs.

But we need consensus from everyone. So I'll let @jmroot and others chime in.

Reply:
> I'm not sure what's the exact intent of the portgroup, but my first impression is that it should handle any ports that install just a README file and maybe add a dependency or two. Nothing else. For those cases I would also expect reasonable defaults for this scenario, and an easy way out (setting a single variable to `true` or `false`) in case the port needs something else.

The exact intent of the portgroup is an important question. There are two main types of stub port:
 1. The kind that exists for user convenience, depends on one or more other ports, and has no dependents. Examples include `gimp`, `texlive`, and all the `p5-*` stubs.
 2. The kind that stands in for another port and has dependents, like `msgpack`.

Type 1 can always be noarch. Type 2 _must not_ be noarch unless all its dependencies are also noarch, because arch checking will not work correctly for the dependents otherwise. (The indirection provided by this kind of stub breaks the rule that ports should depend directly on everything that they use directly.)

If the same portgroup is to be used for both types of stub port, then it must default to not-noarch. Forgetting to make a type 2 stub not-noarch leads to fairly serious breakage. Forgetting to make a stub noarch on the other hand has minimal consequences (two archives per OS version might unnecessarily be built.)

I don't see any benefit to having an extra portgroup option controlling whether the stub is noarch; you can just as easily write `supported_archs noarch` as `stub.noarch yes`.

I can't actually think of any situation where it would be useful to change `installs_libs` from its default value in a stub port. So `stub.libs` can also go.

Reply:
> The exact intent of the portgroup is an important question.

The goals of the pg are two-fold:
* Eliminate the copy-pasta related to stub creation, making it trivial to create on. (Kinda obvious, but...)
* Provide a formal semantic marker - akin to PG `obsolete` - ensuring we can quickly find such ports in a simple and full-proof manner.

> I don't see any benefit to having an extra portgroup option controlling whether the stub is noarch; you can just as easily write `supported_archs noarch` as `stub.noarch yes`.
> 
> I can't actually think of any situation where it would be useful to change `installs_libs` from its default value in a stub port. So `stub.libs` can also go.

IMHO, I prefer to abstract such details via the two formal PG options, `stub.noarch` and `stub.libs`: If there is ever a need/desire to refine the behavior of those, this ensures we can easily do so via the PG. And while I can't think of a concrete example at this point, there may be in the future.

So why not leave them in?

Reply:
> IMHO, I prefer to abstract such details via the two formal PG options, `stub.noarch` and `stub.libs`: If there is ever a need/desire to refine the behavior of those, this ensures we can easily do so via the PG. And while I can't think of a concrete example at this point, there may be in the future.

It might be useful to add a third option like `port.leaf` though, which provides a shortcut to setting `stub.noarch yes` and `stub.libs no`. Previously I had that, via `stub.lib_dep`... but `stub.leaf` might be a better name.

Thoughts?

Reply:
> So why not leave them in?

If you can't think of a single use for them, the better question is why add them in the first place?

Reply:
> > So why not leave them in?
> 
> If you can't think of a single use for them, the better question is why add them in the first place?

If folks don't think they're beneficial, I'm fine with eliminating options `stub.noarch` and `stub.libs`.

> I can't actually think of any situation where it would be useful to change `installs_libs` from its default value in a stub port. So `stub.libs` can also go.

So just to confirm, are you advocating that stubs never set `installs_libs no` any longer?

Reply:
> So just to confirm, are you advocating that stubs never set `installs_libs no` any longer?

Yes. In the cases where it would be OK, it has no effect, because the stub has no dependents or is noarch.

Reply:
> > So just to confirm, are you advocating that stubs never set `installs_libs no` any longer?
> 
> Yes. In the cases where it would be OK, it has no effect, because the stub has no dependents or is noarch.

Works for me. I'll update the PG, and reconcile this PR.

Reply:
> > > So just to confirm, are you advocating that stubs never set `installs_libs no` any longer?
> > 
> > Yes. In the cases where it would be OK, it has no effect, because the stub has no dependents or is noarch.
> 
> Works for me. I'll update the PG, and reconcile this PR.

Changes pushed.